File: src/Core/Quantity.jl
function quantity(concentration::Matrix{Float64}, disc::Discretization)
    w, h = disc.size
    return w * h * sum(concentration) / length(concentration)
end

function reagent_quantity(concentrations::SolutionState, disc::Discretization)
    q_c1 = quantity(concentrations[1], disc)
    q_c2 = quantity(concentrations[2], disc)
    return q_c1 + q_c2
end

File: cpp/src/Quantity.cpp
//
// Created by arnas on 4/23/2026.
//

#include "Quantity.h"

namespace yag_model
{
    double quantity(xt::xarray<double> const& c, Discretization const& disc)
    {
        return xt::sum(c)() * disc.dx * disc.dy;
    }

    double reagentQuantity(SolutionState const& state, Discretization const& disc)
    {
        double const q1 = quantity(state.c1, disc);
        double const q2 = quantity(state.c2, disc);
        return q1 + q2;
    }
}
